packages/host/tests/acceptance/interact-submode-test.gts

```diff
@@ -850,6 +858,76 @@ module('Acceptance | interact submode tests', function (hooks) {
           .doesNotExist('"..." menu does not exist');
       });
     });
+
+    module(
+      'when permission is updated after user has obtained the token',
+      function (hooks) {
+        hooks.beforeEach(async function () {
+          realmPermissions = { [testRealmURL]: ['read'] };
+        });
+
+        test('retrieve a new JWT if recevive 403 error', async function (assert) {
+          let token = createJWT(
+            {
+              user: '@testuser:staging',
+              realm: testRealmURL,
+              permissions: ['read', 'write'],
+            },
+            '7d',
+            testRealmSecretSeed,
+          );
+          let session = {
+            [`${testRealmURL}`]: token,
+          };
+          window.localStorage.setItem('boxel-session', JSON.stringify(session));
+
+          await visitOperatorMode({
+            stacks: [
+              [
+                {
+                  id: `${testRealmURL}Pet/mango`,
+                  format: 'isolated',
+                },
+              ],
+            ],
+          });
+
+          // Mock `Authorization` error response from the server.
+          onFetch = (req, _body) => {
+            if (
+              req.method !== 'GET' &&
+              req.method !== 'HEAD' &&
+              req.headers.get('Authorization') === token
+            ) {
+              return new Response(`Authorization error`, {
+                status: 403,
```

Comment:
I'm open for any other approaches. However, I didn't find a way to trigger `this.checkPermission` to be executed for `testRealm`. Since we're not fetching to `testRealm` trough `realmServer`, instead we're fetching it to `maybeHandle` (https://github.com/cardstack/boxel/blob/21f07d75137958b1f0312fd148b7ab10e2f80a11/packages/runtime-common/realm.ts#L789) from the loader, the request will be treated as local fetch so the `checkPermission` will not be executed.

Reply:
so i guess the point of this PR (and test) is that we are testing new browser functionality. that should mean that we are free to manipulate the server as need be in order to get the client to respond in a way that triggers the new behavior. Given that, I'm ok with us adding/manipulating loader middleware to return the 403 that triggers our new browser logic.